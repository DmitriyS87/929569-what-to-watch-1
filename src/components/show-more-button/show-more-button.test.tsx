import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ShowMoreButton from './show-more-button';
import mockData from '../../mocks/test-mocks.js';

it(`ShowMoreButton correctly renders after relunch with movies.length > length:`, () => {
  const tree = renderer.create(
    <ShowMoreButton
      movies={mockData.films}
      limit={2}
      setNewShowLimit={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`ShowMoreButton correctly renders after relunch with movies.length <= length:`, () => {
  const tree = renderer.create(
    <ShowMoreButton
      movies={mockData.films}
      limit={3}
      setNewShowLimit={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
