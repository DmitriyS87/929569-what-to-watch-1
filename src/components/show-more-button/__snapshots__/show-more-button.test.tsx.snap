// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton correctly renders after relunch with movies.length <= length: 1`] = `<div />`;

exports[`ShowMoreButton correctly renders after relunch with movies.length > length: 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
